Add geo position form

akechahmadze opened the pull request:
Форма для задания локации

k-teplyakov commented on .gitignore:
```diff
@@ -18,3 +18,23 @@
 npm-debug.log*
```
давай удалим из репы `build` и добавим ее сюда

k-teplyakov commented on src/apiKey.ts:
```diff
@@ -0,0 +1 @@
+export const n2yoApiKey = 'BFSKBN-KA5QSC-7TKTJY-4UB7'
```
давай это вынесем в `.env` и подключим либу `dotenv`

k-teplyakov commented on src/Requests/above.ts:
```diff
@@ -0,0 +1,32 @@
+    category: number;
+}
+
+export const getAboveData = async ({ latitude, longitude, radius, category }: UserFormData): Promise<AboveData> => {
```
давай переименуем папку `Requests` в `api` и будем здесь группировать функции по источнику

k-teplyakov commented on src/RouteComponents/Main/index.tsx:
```diff
@@ -0,0 +1,12 @@
+import { LocationForm } from '../../Components/LocationForm'
+import './index.scss'
+
+export const Main = () => {
```
давай переименуем `RouteComponents` в `components`
И договоримся о том, что с большой буквы будем писать только имена файлов, которые содержат имена компонентов

akechahmadze replied:
@k-teplyakov 
Ок, а сопутствующий файл `index.sccs` как обозвать? А другие файлы, которые рядом лежат? Например, у компонента `LocationForm` рядом есть `validationScheme.ts` и `satelliteCategories.ts`

k-teplyakov replied:
сопутствующие файлы в нижнем регистре по назначению. например, `scheme.ts` `constants.ts` `index.sccs` etc

akechahmadze replied:
@k-teplyakov  То есть будет выглядеть вот так?
- components
    - Main
        - Main.tsx
        - index.scss
     - SatelliteInfo
         - SatelliteInfo.tsx

Но тогда импорты будут `components/Main/Main`

k-teplyakov replied:
- components
    - main
        - index.tsx
        - index.scss
        - UniqeButton.tsx
     - satelliteInfo
         - index.tsx
         - AdditionalComponent.tsx

akechahmadze replied:
@k-teplyakov я запутался, а что тогда с большой буквы? )

k-teplyakov commented on src/Components/LocationForm/validationScheme.ts:
```diff
@@ -0,0 +1,8 @@
+
+export const scheme = object({
+  latitude: number().required().min(0),
+  longitude: number().required().min(0),
```
а у них нет максимального значения?

k-teplyakov commented on src/Components/LocationForm/index.tsx:
```diff
@@ -0,0 +1,75 @@
+import { yupResolver } from '@hookform/resolvers/yup'
+import { scheme } from './validationScheme'
+import { satelliteCategories } from './satelliteCategories'
+import { satelliteTableStore } from '../../Stores/SatelliteTableStore'
```
давай использовать абсолютные пути

k-teplyakov commented on src/Components/LocationForm/validationScheme.ts:
```diff
@@ -0,0 +1,8 @@
+  latitude: number().required().min(0),
+  longitude: number().required().min(0),
+  radius: number().required().min(0).max(90),
+  category: number().required()
```
наверняка здесь тоже можно проверить валидность категории)

k-teplyakov commented on src/Components/LocationForm/satelliteCategories.ts:
```diff
@@ -0,0 +1,12 @@
+    id: number;
+}
+
+export const satelliteCategories: SatelliteCategory[] = [
```
по идее эта информация является частью `api` сервиса, давай туда ее и перенесем

akechahmadze replied:
@k-teplyakov  Почему? По этим данным мы рисуем селект. А сервис принимает лишь `id` и ничего не знает про имя

k-teplyakov replied:
ну в теории, этот набор данных полностью привязан к api, это их категории
и должен был бы отдаваться каким-то методом апихи, но они его просто опубликовали таблицей

akechahmadze committed:
refactor: required changes

# Conflicts:
#	src/api/above.ts

akechahmadze committed:
refactor: imports

# Conflicts:
#	src/Components/Main/index.tsx
#	src/Stores/SatelliteTableStore.ts

akechahmadze committed:
Merge branch 'main' of https://github.com/akechahmadze/satellitesaboveme into form

# Conflicts:
#	.eslintrc.js
#	src/Components/Main/index.tsx
#	src/Stores/SatelliteTableStore.ts

akechahmadze committed:
refactor: required changes
